### DISCUSSION TITLE
certificate generate issue for Nishtha program.

### DISCUSSION THREAD

**Discussion by kamal-singhal** [Sunbird-Lern/Community#127] [Issues] [answered]
Hi team, 

User has raised a issue that he completed the course under nishtha program but yet not received the certificate for the same. can you help us for the same. we are trying but not get solution. we have checked the criteria of course and user has fulfilled the criteria.
1. can you tell us debugging steps.
2.  which job has responsible to generate the certification after completion. 
3.  once certificate generate, in which db gets stored and also let us know in which table.

**Comment by reshmi-nair** [comment, marked as answer]
@kamal-singhal Please refer to below link for certificate generation flow 
[certificate generation](https://lern.sunbird.org/learn/product-and-developer-guide/data-pipeline-flink-jobs/datapipeline-flink-jobs)

Also check the collection-cert-pre-processor and collection-certificate-generator flink jobs (task manager logs) for the specific course id and batch id, to see if the events are processed. collection-certificate-generator calls sunbird-RC to generate certificates.. So please check if registry, certificate-api and certificate-sign services is running and logs.

Certificate is stored in RC postgres database - registry schema - V_TrainingCertificate table.

**Reply by kamal-singhal** [reply]
Thank you for your response. Really it is very help us.

**Reply by Jayaprakash8887** [reply]
@kamal-singhal Please mark this thread as answered if there are no further queries.
